templates/generateGptQuery.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
+
+# Define process input variables
+data_path = "${data}"
+source = "${source}"
+target_index = "${index}"
+target_index = int(target_index)
+num_genes = "${count}"
+num_genes = int(num_genes)
+mode = "${mode}"
+question = "${question}"
+
+# Open data file
+with open(data_path, "r") as file:
+    # Read the header row (though it’s not necessary for column names anymore)
+    header = file.readline().strip().split("\t")
+
+    # Ensure target_index is within the bounds of available columns
+    if target_index >= len(header) or target_index < 0:
+        print(f"Error: The specified column index {target_index} is out of range!")
```

Comment:
is this expected to fail? we should throw an error if so, this will only print the message but continue the execution of the script